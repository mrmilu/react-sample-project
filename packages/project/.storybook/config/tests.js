import { addDecorator } from '@storybook/react';
import { withTests } from '@storybook/addon-jest';

import results from '../../src/stories/.jest-test-results.json';

addDecorator(withTests({ results }));
