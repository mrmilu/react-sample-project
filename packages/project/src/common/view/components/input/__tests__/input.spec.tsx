import React, { useState } from 'react';
import { customRender, fireEvent, mockComponent } from '../../../../../../tests/test-utils';
import Input from '../input.component';

jest.mock('../../title/title.component', () => mockComponent('Title'));

function TestInput() {
  const [value, setValue] = useState('');

  return (
    <Input
      label="Test"
      onChange={(e) => {
        setValue(e.target.value);
      }}
      name="test"
      value={value}
    />
  );
}

const setup = () => {
  const rendered = customRender(<TestInput />);
  const input = rendered.getByLabelText('Test') as HTMLInputElement;
  return {
    input,
    rendered
  };
};

describe('Input', () => {
  it('Render', () => {
    const { rendered } = setup();
    // Generate a master snapshot or check if already exists a master
    expect(rendered.container).toMatchSnapshot();
  });

  it('Should trigger on change', () => {
    const { input } = setup();
    fireEvent.change(input, { target: { value: 'This is a test' } });
    expect(input.value).toBe('This is a test');
  });
});
