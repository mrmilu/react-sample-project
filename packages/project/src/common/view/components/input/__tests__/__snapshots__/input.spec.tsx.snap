// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Render 1`] = `
<div>
  <div>
    <label
      for="test"
    >
      Test
    </label>
    <input
      id="test"
      style="color: black;"
      type="text"
      value=""
    />
  </div>
</div>
`;
